src/Hamiltonians/TwoBodyRealCorrelation.jl

```diff
@@ -0,0 +1,62 @@
+"""
+    G2RealCorrelator(d::Int) <: AbstractHamiltonian{Float64}
+
+Two-body operator for density-density correlation between sites separated by `d`.
+```math
+    G_2(d) = \\sum_i^M \\hat{n}_i (\\hat{n}_{i+d} - \\delta_{0d}).
```

Comment:
Just to keep notation consistent with `G2Correlator`:
```suggestion
    \\hat{G}^{(2)}(d) = \\sum_i^M \\hat{n}_i (\\hat{n}_{i+d} - \\delta_{0d}).
```